astropy/coordinates/tests/test_earth.py

```diff
@@ -75,19 +74,19 @@ def test_gd2gc():
     status = 0  # help for copy & paste of vvd
 
     location = EarthLocation.from_geodetic(e, p, h, ellipsoid='WGS84')
-    xyz = location.to_geocentric()
+    xyz = tuple(v.to(u.m) for v in location.to_geocentric())
     vvd(xyz[0], -5599000.5577049947, 1e-7, "eraGd2gc", "0/1", status)
     vvd(xyz[1], 233011.67223479203, 1e-7, "eraGd2gc", "1/1", status)
     vvd(xyz[2], -3040909.4706983363, 1e-7, "eraGd2gc", "2/1", status)
 
     location = EarthLocation.from_geodetic(e, p, h, ellipsoid='GRS80')
-    xyz = location.to_geocentric()
+    xyz = tuple(v.to(u.m) for v in location.to_geocentric())
     vvd(xyz[0], -5599000.5577260984, 1e-7, "eraGd2gc", "0/2", status)
     vvd(xyz[1], 233011.6722356703, 1e-7, "eraGd2gc", "1/2", status)
     vvd(xyz[2], -3040909.4706095476, 1e-7, "eraGd2gc", "2/2", status)
 
     location = EarthLocation.from_geodetic(e, p, h, ellipsoid='WGS72')
-    xyz = location.to_geocentric()
+    xyz = tuple(v.to(u.m) for v in location.to_geocentric())
```

Comment:
`allclose` is used in `vvd`. Before, I relied on values from `to_geocentric` being already in `m` and the unit being ignored; now it explicitly converts to `m` (we might want to change the default, who knows...) and in `vvd` attaches that unit to the comparison (which is inserted using cut & paste input, and which I'd like not to touch too much).